File: BlinkyLights/blinkOnce.java
/*
 * blinkOnce.java
 *
 * Control a blinking LED with RPi
 *
 */
import com.pi4j.io.gpio.*;
import com.pi4j.io.gpio.trigger.GpioCallbackTrigger;

/* Define a function to control the LED */
public class blinkOnce {
    private static void blink(GpioPinDigitalOutput led, int time, int dur){
        led.blink(time, dur);
        return;
    }

    public static void main(String [] args){
        System.setProperty("pi4j.linking", "dynamic");
        
        // Initialize the GPIO
        GpioFactory.setDefaultProvider(new RaspiGpioProvider(RaspiPinNumberingScheme.BROADCOM_PIN_NUMBERING));
        final GpioController gpio = GpioFactory.getInstance();
        final GpioPinDigitalOutput led = gpio.provisionDigitalOutputPin(RaspiPin.GPIO_17);
    
        // Run the function in a loop
        try {
            System.out.println("The LED will blink ten times");
            for(int cntr=0; cntr<10; cntr++){
                System.out.printf("Blink #%d\n", cntr+1);
                blink(led, 1000, 500);
                Thread.sleep(2000);
            }
        } catch (InterruptedException e){
            gpio.shutdown();
        }
    }
}
            

File: BlinkyLights/blinkOnce.py
#!/usr/bin/env python

#
# blinkOnce.py
#
# Control a blinking LED with RPi
#

import RPi.GPIO as GPIO
import time

# Initialize the GPIO
GPIO.setmode(GPIO.BCM)
GPIO.setup(17,GPIO.OUT)

# Define a function to control the LED
def blinkOnce(pin):
	GPIO.output(pin,True)
	time.sleep(.5)
	GPIO.output(pin,False)
	time.sleep(.5)

# Run the function in a loop
try:
	for i in range(10):
		blinkOnce(17)
except KeyboardInterrupt:
	print "Bye"
	# Cleanup
	GPIO.cleanup()
